src/microbe_stage/Microbe.cs

```diff
@@ -1426,23 +1426,61 @@ private void OnIGotAddedToColony()
         if (ColonyParent == null)
             return;
 
-        var oldRotation = Rotation;
-        var vectorToParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
+        var newTransform = GetNewRelativeTransform();
+
+        Rotation = newTransform.rotation;
+        Translation = newTransform.translation;
+
         ChangeNodeParent(ColonyParent);
+    }
+
+    /// <summary>
+    ///   This method calculates the relative rotation and translation this microbe should have to his microbe parent.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Storing the old global translation and rotation, re-parenting and then reapplying the stored values is not
+    ///     worse than this code because this code utilizes GetVectorTowardsNearestPointOfMembrane. This reduces the
+    ///     visual gap between the microbes in a colony.
+    ///   </para>
+    /// </remarks>
+    /// <returns>Returns relative translation and rotation</returns>
+    private (Vector3 translation, Vector3 rotation) GetNewRelativeTransform()
+    {
+        // Gets the global rotation of the parent
+        var globalParentRotation = ColonyParent.GlobalTransform.basis.GetEuler();
+
+        // A vector from the parent to me
+        var vectorFromParent = GlobalTransform.origin - ColonyParent.GlobalTransform.origin;
+
+        // A vector from me to the parent
+        var vectorToParent = -vectorFromParent;
+
+        // This vector represents the vectorToParent as if I had no rotation.
+        // This works by rotating vectorToParent by the negative value (therefore Down) of my current rotation
+        // This is important, because GetVectorTowardsNearestPointOfMembrane only works with non-rotated microbes
+        var vectorToParentWithoutRotation = vectorToParent.Rotated(Vector3.Down, Rotation.y);
```

Comment:
Why not use the quaternion rotation, instead decomposing (and losing information) first and losing information. then requiring to specify the direction here.
Then again I'm having trouble understanding what goes here?
What does this actually do?

If we have a direction vector in local coordinates towards the parent, shouldn't it be enough to just discard the Y value and then use x and z as the x and y values to get to organelle position coordinates to use with the membrane code?

Reply:
Perhaps drawing a picture (I can upload one somewhere to link here) would greatly improve the understandability of this code.

Reply:
I've made some scrambles on a piece of paper, but it isn't really that useful. I could try to visualize this bettet once I'm back

Reply:
Maybe this helps a little bit... (There is a reason why I'm not in the graphics team :p)
![VectorExpl](https://user-images.githubusercontent.com/30809803/129459218-c67feff3-1639-4c44-be3c-da6ea31ffda6.png)

Raw Gimp file: [VectorExpl.zip](https://github.com/Revolutionary-Games/Thrive/files/6987151/VectorExpl.zip)

Reply:
Uploaded that image to storage that github won't delete at some point: https://randomthrivefiles.b-cdn.net/documentation/fixed_colony_rotation_explanation_image.png